src/pydvl/value/shapley/montecarlo.py

```diff
@@ -152,6 +148,146 @@ def permutation_montecarlo_shapley(
     return map_reduce_job()
 
 
+def permutation_montecarlo_classwise_shapley(
+    u: Utility,
+    label: int,
+    *,
+    done: StoppingCriterion,
+    truncation: TruncationPolicy,
+    use_default_scorer_value: bool = True,
+    min_elements_per_label: int = 1,
+) -> ValuationResult:
+    """
+    Samples a random subset of the complement set and computes the truncated Monte Carlo
+    estimator.
+
+    :param u: Utility object containing model, data, and scoring function. The scoring
+        function should be of type :class:`~pydvl.utils.score.ClassWiseScorer`.
+    :param done: Function checking whether computation needs to stop.
+    :param label: The label for which to sample the complement (e.g. all other labels)
+    :param truncation: Callable which decides whether to interrupt processing a
+        permutation and set all subsequent marginals to zero.
+    :param use_default_scorer_value: Use default scorer value even if additional_indices
+        is not None.
+    :param min_elements_per_label: The minimum number of elements for each opposite
+        label.
+    :return: ValuationResult object containing computed data values.
+    """
+
+    algorithm_name = "classwise_shapley"
+    result = ValuationResult.zeros(
+        algorithm="classwise_shapley",
+        indices=u.data.indices,
+        data_names=u.data.data_names,
+    )
+
+    _, y_train = u.data.get_training_data(u.data.indices)
+    class_indices_set, class_complement_indices_set = split_indices_by_label(
+        u.data.indices,
+        y_train,
+        label,
+    )
+    _, complement_y_train = u.data.get_training_data(class_complement_indices_set)
+    indices_permutation = np.random.permutation(class_indices_set)
+
+    for subset_idx, subset_complement in enumerate(
+        random_powerset_group_conditional(
+            class_complement_indices_set,
+            complement_y_train,
+            min_elements_per_group=min_elements_per_label,
+        )
+    ):
+        result += _permutation_montecarlo_shapley_rollout(
+            u,
+            indices_permutation,
+            additional_indices=subset_complement,
+            truncation=truncation,
+            algorithm_name=algorithm_name,
+            use_default_scorer_value=use_default_scorer_value,
+        )
+        if done(result):
```

Comment:
Is checking here (in the inner, class-conditional loop) the right thing? Values undergo further scaling and weighting after this, correct?

Reply:
It is correct. The done criterion for this specific function is actually not the outer one. It got replaced by `MaxChecks(n_resample_complement_sets)`. To ensure that exactly `n_resample_complement_sets` complement sets are sampled in the function, this call is necessary.

As it wasn't directly clear, do you think it is better to remove the done parameter and replace by the int value directly. And subsequently replace the check with `idx==n_resample_complement_sets`.